File: lib/src/main/java/net/llvg/loliutils/utils/CompactEnumeration.java
package net.llvg.loliutils.utils;

import java.util.*;

@SuppressWarnings ("unused")
public class CompactEnumeration<T> implements Enumeration<T> {
        private final Enumeration<T>[] enums;
        private int index = 0;
        
        public CompactEnumeration(Enumeration<T>[] enums) {
                this.enums = enums;
        }
        
        @Override public boolean hasMoreElements() {
                while (index < enums.length) {
                        if (enums[index].hasMoreElements()) {
                                return true;
                        }
                        ++index;
                }
                return false;
        }
        
        @Override public T nextElement() {
                if (hasMoreElements()) {
                        return enums[index].nextElement();
                }
                
                throw new NoSuchElementException();
        }
}

File: lib/src/main/kotlin/net/llvg/loliutils/utils/CompactEnumeration.kt
package net.llvg.loliutils.utils

import java.util.Enumeration

@Suppress("UNUSED")
class CompactEnumeration<T>(private vararg val enums: Enumeration<T>) : Enumeration<T> {
        private var index = 0
        override fun hasMoreElements(): Boolean {
                while (index < enums.size) {
                        if (enums[index].hasMoreElements()) return true
                        ++index
                }
                return false
        }
        
        override fun nextElement(): T {
                if (hasMoreElements()) return enums[index].nextElement()
                
                throw NoSuchElementException()
        }
}